x-pack/plugins/alerts/server/alert_type_registry.ts

```diff
@@ -137,3 +142,26 @@ function normalizedActionVariables(actionVariables: AlertType['actionVariables']
     params: actionVariables?.params ?? [],
   };
 }
+
+function validateActionGroups(alertTypeId: string, actionGroups: ActionGroup[]) {
+  const reservedActionGroups = intersection(
+    actionGroups.map((item) => item.id),
+    getBuiltinActionGroups().map((item) => item.id)
+  );
+  if (reservedActionGroups.length > 0) {
+    throw Boom.badRequest(
+      i18n.translate('xpack.alerts.alertTypeRegistry.register.reservedActionGroupUsageError', {
+        defaultMessage:
+          'Alert type by id {alertTypeId} cannot be registered. Action groups [{actionGroups}] is reserved by framework.',
+        values: {
+          actionGroups: reservedActionGroups.join(', '),
+          alertTypeId,
+        },
+      })
+    );
+  }
+}
+
+function getBuiltinActionGroups(): ActionGroup[] {
```

Comment:
nit: Would this function make more sense inside `x-pack/plugins/alerts/common/builtin_action_groups.ts`?